Automattic/wp-calypso
# Gutenboarding: Remove all unused bits and reorganize the rest

While working on #74399 I noticed that none of Gutenboarding is accessible, since it would always redirect to `/start`:

https://github.com/Automattic/wp-calypso/blob/3626bab4dd8d7785adc0b13fa501499220be1774/client/server/pages/index.js#L901-L915

Regardless that we won't be using it, we are building the Gutenboarding entry point along with all of its code.

Might be easier to review on a per-commit basis, since all changes that aren't removals are in separate commits.

## Proposed Changes

This PR removes the unused Gutenboarding entry point, along with all unused code from there.

The little remaining code is being moved where it is currently used and makes more sense.

## Testing Instructions

* Verify that you can't reach any Gutenboarding route (`/new*`)
* Smoke test new user signup and existing user site creation flow. 
* Verify all checks are green.
